File: reorder.js
const fs = require('fs');
const content = fs.readFileSync('index.html', 'utf8');

let newContent = content.replace(/\s*<h3>Client — B Studio<\/h3>/g, '');

const startMarker = '<h1 id=\"projects\">Projects</h1>';
const startIndex = newContent.indexOf(startMarker);
if (startIndex !== -1) {
    const hrIndex = newContent.indexOf('<hr />', startIndex);
    const endMarker = '<p class=\"scroll-animate fade-in-up\">';
    const endIndex = newContent.indexOf(endMarker, hrIndex);
    
    let projectsBlock = newContent.substring(hrIndex, endIndex);
    let parts = projectsBlock.split('<hr />');
    
    let topParts = [];
    let bottomParts = [];
    
    for (let i = 1; i < parts.length; i++) {
        if (!parts[i].trim()) continue;
        if (parts[i].includes('Eurafatours') || parts[i].includes('Camp Abdelmoula') || parts[i].includes('Defi du Desert') || parts[i].includes('The Untold Layers')) {
            topParts.push(parts[i]);
        } else {
            bottomParts.push(parts[i]);
        }
    }
    
    let newProjectsBlock = '\         <hr />' + topParts.join('<hr />') + bottomParts.join('<hr />') + '\\         ';
    
    newContent = newContent.substring(0, hrIndex) + newProjectsBlock + newContent.substring(endIndex);
    fs.writeFileSync('index.html', newContent);
    console.log('Success');
} else {
    console.log('Start marker not found');
}

File: reorder.py
import re

with open('index.html', 'r', encoding='utf-8') as f:
    html = f.read()

# Find the projects section
# Everything between <h1 id="projects">Projects</h1><hr /> and <p >\s*These projects represent
pattern = re.compile(r'(<h1 id="projects">Projects</h1>\s*<hr />\s*)(.*?)\s*(<p(?:[^>]*)>\s*These projects represent)', re.DOTALL)

match = pattern.search(html)
if match:
    prefix = match.group(1)
    projects_content = match.group(2)
    suffix = match.group(3)
    
    # split projects_content by <hr />
    parts = re.split(r'\s*<hr />\s*', projects_content)
    projects = [p.strip() for p in parts if p.strip()]
    
    # find the specific projects
    eurafatours = [p for p in projects if 'Eurafatours &amp; Domaine des Citronniers' in p]
    camp = [p for p in projects if 'Camp Abdelmoula' in p]
    defi = [p for p in projects if 'Défi du Désert Voyages' in p]
    untold = [p for p in projects if 'The Untold Layers' in p]
    
    others = [p for p in projects if p not in eurafatours + camp + defi + untold]
    
    # Put the 3 requested ones at the top, followed by untold (since it's a B-studio project too, usually grouped), then others
    ordered_projects = eurafatours + camp + defi + untold + others
    
    new_projects_content = '\n         <hr />\n         '.join(ordered_projects)
    
    # replace back
    new_html = html[:match.start(2)] + new_projects_content + '\n\n         ' + html[match.start(3):]
    
    with open('index.html', 'w', encoding='utf-8') as f:
        f.write(new_html)
    print("Successfully reordered projects in index.html")
else:
    print("Could not find the projects section using regex.")
